src/classes/TestResult.js

```diff
@@ -142,20 +142,24 @@ export default class TestResult extends BubblingEventTarget {
 
 		this.tests = this.test.tests?.map(t => new TestResult(t, this, childOptions));
 
-		delay(1).then(() => {
-			if (this.test.isTest) {
-				if (this.test.skip) {
-					this.skip();
-				}
-				else {
-					Promise.resolve(this.test.setup())
-						.then(() => this.run())
-						.finally(() => this.test.teardown());
+		delay(1)
+			.then(() => Promise.resolve(this.test.beforeAll?.()))
```

Comment:
Does it make sense to limit `beforeAll`/`afterAll` to groups only, for example, like so?

```js
Promise.resolve(this.isGroup && this.test.beforeAll?.())
```

Reply:
Why?

Reply:
Btw you don't need `Promise.resolve()` here I believe, anything will be wrapped with a promise anyway.

Reply:
> Btw you don't need `Promise.resolve()` here I believe, anything will be wrapped with a promise anyway.

Oh, right. Thank you!